qcfractal/interface/collections/reaction_dataset.py

```diff
@@ -92,9 +95,58 @@ def _form_index(self):
         self.rxn_index = pd.DataFrame(tmp_index, columns=["name", "stoichiometry", "molecule", "coefficient"])
         self.valid_stoich = set(self.rxn_index["stoichiometry"].unique())
 
+    def _molecule_indexer(self,
+                          stoich: Union[str, List[str]],
+                          subset: Optional[Union[str, Set[str]]] = None,
+                          coefficients: bool = False) -> Tuple[Dict[Tuple[str, ...], 'ObjectId'], Tuple[str]]:
+        """Provides a {index: molecule_id} mapping for a given subset.
+
+        Parameters
+        ----------
+        stoich : Union[str, List[str]]
+            The stoichiometries, or list of stoichiometries to return
+        subset : Optional[Union[str, Set[str]]], optional
```

Comment:
`subset` is not used in this function.

Reply:
Whoops, this was the heart of the issue. Fixed.